File: get-started/quickstart/java/app/src/main/java/kiotaposts/client/PostsClient.java
package kiotaposts.client;

import com.microsoft.kiota.ApiClientBuilder;
import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import com.microsoft.kiota.serialization.ParseNodeFactoryRegistry;
import com.microsoft.kiota.serialization.SerializationWriterFactoryRegistry;
import java.util.HashMap;
import java.util.Objects;
import kiotaposts.client.posts.PostsRequestBuilder;
/**
 * The main entry point of the SDK, exposes the configuration and the fluent API.
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class PostsClient extends BaseRequestBuilder {
    /**
     * The posts property
     * @return a {@link PostsRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public PostsRequestBuilder posts() {
        return new PostsRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * Instantiates a new {@link PostsClient} and sets the default values.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public PostsClient(@jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}");
        this.pathParameters = new HashMap<>();
        if (requestAdapter.getBaseUrl() == null || requestAdapter.getBaseUrl().isEmpty()) {
            requestAdapter.setBaseUrl("https://jsonplaceholder.typicode.com");
        }
        pathParameters.put("baseurl", requestAdapter.getBaseUrl());
    }
}

File: get-started/quickstart/cli/src/Client/PostsClient.cs
using KiotaPostsCLI.Client.Posts;
using Microsoft.Kiota.Abstractions;
using Microsoft.Kiota.Abstractions.Extensions;
using Microsoft.Kiota.Cli.Commons;
using Microsoft.Kiota.Cli.Commons.IO;
using Microsoft.Kiota.Serialization.Form;
using Microsoft.Kiota.Serialization.Json;
using Microsoft.Kiota.Serialization.Text;
using System;
using System.Collections.Generic;
using System.CommandLine;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace KiotaPostsCLI.Client {
    /// <summary>
    /// The main entry point of the SDK, exposes the configuration and the fluent API.
    /// </summary>
    public class PostsClient : BaseCliRequestBuilder {
        /// <summary>
        /// The posts property
        /// </summary>
        public Command BuildPostsCommand() {
            var command = new Command("posts");
            command.Description = "The posts property";
            var builder = new PostsRequestBuilder(PathParameters);
            command.AddCommand(builder.BuildCommand());
            command.AddCommand(builder.BuildCreateCommand());
            command.AddCommand(builder.BuildListCommand());
            return command;
        }
        /// <summary>
        /// Instantiates a new PostsClient and sets the default values.
        /// </summary>
        public Command BuildRootCommand() {
            var command = new RootCommand();
            command.Description = "Instantiates a new PostsClient and sets the default values.";
            command.AddCommand(BuildPostsCommand());
            return command;
        }
        /// <summary>
        /// Instantiates a new PostsClient and sets the default values.
        /// </summary>
        public PostsClient() : base("{+baseurl}", new Dictionary<string, object>()) {
        }
    }
}
